babel/test/unit/babel_client_test.js

```diff
@@ -1109,75 +1027,51 @@ describe("Babel Node Client Test Suite", function(){
             createAnnotation.should.throw("Invalid data: hasTarget has unrecognised property 'something'");
         });
         it("- should return an error (401) if persona token is invalid", function(done){
-            var babel = rewire("../../index.js");
-
             var babelClient = babel.createClient({
                 babel_host:"http://babel",
                 babel_port:3000
             });
-            var requestStub = {
-                post:function(options, callback){
-                    options.headers.should.have.property('User-Agent', 'talis-node/0.2.1');
-                    var error = new Error('The token is invalid or has expired');
-                    error.http_code = 401;
-                    callback(error);
-                }
-            };
-
-            babel.__set__("request", requestStub);
+            nock(endPoint)
+                .post('/annotations')
+                .reply(function(uri, requestBody){
+                    this.req.headers['user-agent'].should.equal('talis-node/0.2.3');
+                    return [401, {
+                        error:"invalid_token",
+                        error_description:"The token is invalid or has expired"
+                    }];
+                });
 
             babelClient.createAnnotation('secret', {hasBody:{format:'text/plain', type:'Text'}, hasTarget:{uri:'http://example.com'}, annotatedBy:'Gordon Freeman'},  {}, function(err, result){
 
                 (err === null).should.be.false;
                 err.http_code.should.equal(401);
-                err.message.should.equal('The token is invalid or has expired');
-                (typeof result).should.equal('undefined');
-                done();
-            });
-        });
-
-        it("- should return an error if call to request returns an error", function(done){
-            var babel = rewire("../../index.js");
-
-            var babelClient = babel.createClient({
-                babel_host:"http://babel",
-                babel_port:3000
-            });
-            var requestStub = {
-                post:function(options, callback){
-                    options.headers.should.have.property('User-Agent', 'talis-node/0.2.1');
-                    var error = new Error('Error communicating with Babel');
-                    callback(error);
-                }
-            };
+                // TODO This is a change - but all the other error handling tests are like this
+                // if I make this one pass without change - then all the others fail. 
```

Comment:
The error handling here has changed. But changing how the library formats the error response here, breaks other tests - something has to change!

I don't like the error handling - it is returning a text string with some json in the text - which seems crazy.